### DISCUSSION TITLE
Prompt Engineer Agent for the new AgentFlow

### DISCUSSION THREAD

**Discussion by toi500** [FlowiseAI/Flowise#2571] [Show and tell]
This simple `Prompt Engineer Agent / System` can craft any system instructions you need to build your AgentFlow graph. Simply explain the app you want to create, and it will generate the system messages for each Worker.

**Note :** The more details you provide for your graph, the better results you will get.

### VERSION 1
Flow: 
[Prompt Engineer.json](https://github.com/user-attachments/files/15566889/Prompt.Engineer.json) - Via Chatflow

Example 1:

--
<img width="854" alt="Screenshot 2024-06-04 025816" src="https://github.com/FlowiseAI/Flowise/assets/138339291/4beea501-aab2-45fd-b620-c38a27686ea8">

Example 2, using tools:

--
<img width="719" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/138339291/d8f2752d-dae0-4a5d-b60f-802cf5fe1bea">

### VERSION 2
Flow: 
[Prompt Engineer System.json](https://github.com/user-attachments/files/15566891/Prompt.Engineer.System.json) - Via AgentFlow

Here is an improved version using 2 Prompt Engineer Agents, working together to generate clean worker prompts:

--
<img width="499" alt="image" src="https://github.com/FlowiseAI/Flowise/assets/138339291/66dd18d2-07c4-4674-b19f-f20a0862d77c">

**Comment by toi500** [comment]
@HenryHengZJ, do you think it would be worthwhile to have a flow template on the marketplace based on this concept?

**Reply by HenryHengZJ** [reply]
of course, this is useful!

**Comment by gamersalpha** [comment]
Hello where is the last version all node are outdated thanks

**Reply by DevstackK** [reply]
you can get the latest nodes just search from the flowise menu
